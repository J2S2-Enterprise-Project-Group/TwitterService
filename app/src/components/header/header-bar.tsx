/**
 * @author Shiv Kumar Ganesh <shivkumar.ganesh@sjsu.edu>
 */
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
}));

interface HeaderBarProps {
  title?: string;
}

export const HeaderBar: React.FC<HeaderBarProps> = ({ title = 'Twitter App' }) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6" className={classes.title}>
            {title}
          </Typography>
        </Toolbar>
      </AppBar>
    </div>
  );
}
