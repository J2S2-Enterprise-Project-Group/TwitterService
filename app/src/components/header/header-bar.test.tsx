/**
 * @author : Shiv Kumar Ganesh<shivkumar.ganesh@sjsu.edu>
 */
import React from 'react';
import { shallow } from 'enzyme';
import {HeaderBar} from './header-bar';
import { Typography } from '@material-ui/core';

const title = 'Twitter App';
let wrapped = shallow(<HeaderBar title={title} />);
describe('Header Component', () => {
  it('Check if the component loaded', () => {   
    expect(wrapped).toMatchSnapshot();
  });

  it('renders the Titles children', () => { 
    expect(wrapped.find(Typography).text()).toEqual(title);
  });
});