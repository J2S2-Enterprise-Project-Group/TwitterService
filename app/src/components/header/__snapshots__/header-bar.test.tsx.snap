// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Check if the component loaded 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        variant="h6"
      >
        Twitter App
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
